### PR TITLE
Remove legacy/unused code from lib/wp

### PR DESCRIPTION
**Pull request by jsnajdr**
Removes two bits of legacy code from `lib/wp`:
- setting the `window.wpcom.__version` value. Removes a need to require and bundle a `package.json` module
- the `http-envelope-normalizer` module that's unused. It was used by a "Google voucher" feature 5 years ago (#5765) but that's no longer in the codebase.

### CODE REVIEW

File: client/lib/wp/handlers/http-envelope-normalizer.js [hunk 1]
```diff
@@ -1,45 +0,0 @@
-/**
- * Detect error looking in the reponse data object.
- *
- *
- * @param {Function} handler - wpcom handler
- * @returns {Function} handler wrapper
- */
-
-export function requestHandler( handler ) {
```
Comment (tyxla) [inline client/lib/wp/handlers/http-envelope-normalizer.js:9]: Nice catch that this is no longer used 👍

**Review by tyxla — APPROVED** [review]
Nice cleanup 👍  LGTM! 🚀

File: client/lib/wp/browser.js [hunk 2]
```diff
@@ -38,10 +38,8 @@ if ( config.isEnabled( 'support-user' ) ) {
 if ( 'development' === process.env.NODE_ENV ) {
 	require( './offline-library' ).makeOffline( wpcom );
 
-	// expose wpcom global var only in development
-	const wpcomPKG = require( 'wpcom/package' );
+	// expose wpcom global var in development mode
 	window.wpcom = wpcom;
-	window.wpcom.__version = wpcomPKG.version;
```
Comment (tyxla) [inline client/lib/wp/browser.js:44]: Do you recall why this was necessary?

Comment (jsnajdr) [reply client/lib/wp/browser.js:44]: I don't know. It was added in #4880 a _very_ long time ago, and seems quite unrelated to other changes in that PR.

**Pull request merged**